install-wrappers.sh

```diff
@@ -58,6 +64,45 @@ else
     esac
 fi
 
+if [ -n "${HOST_CLANG-}" ]; then
+    HOST_CLANG_EXE=$(command -v $HOST_CLANG)
+    HOST_CLANG_VER=$(echo "__clang_major__ __clang_minor__ __clang_patchlevel__" | $HOST_CLANG_EXE -E -P -x c - | xargs printf '%d.%d.%d')
+
+    mkdir -p $PREFIX/bin
+
+	# ex. /usr/lib/llvm-17/lib/clang/17
+    resdir=$($HOST_CLANG -print-resource-dir)
+	# ex. /usr/lib/llvm-17
+    llvmdir=${resdir%/lib/clang/*}
+	# ex /lib/clang/17
+    clangres=${resdir#$llvmdir}
+
+    mkdir -p $PREFIX$clangres
+
+    # link the header directory, prevent modification
+    ln -snf $resdir/include $PREFIX$clangres/include
+
+    # Note: clang will detect the "InstalledDir" based on the path that was used to invoke the tools
+    # This might still have some hidden effects
+    printf '#!/bin/sh\nsr=$(dirname "$(dirname "$(readlink -f "$0")")")\nexec %s -resource-dir="$sr"%s --sysroot="$sr" "$@"\n' "$HOST_CLANG_EXE" "$clangres" > $PREFIX/bin/clang
```

Comment:
Why the `-f` to readlink here? Won't that have the effect that in case we're not invoked via a symlink, but directly, that readlink would fail, instead of just silently giving `$0`? But `clang-target-wrapper.sh` still executes `clang` directly, no?

Reply:
`readlink -f` will give you a fully resolved, absolute path.
More consistent considering you could invoke via symlinks or relative paths.

Reply:
Oh, now I see the source of my confusion, I misread the macOS manpage. It says
```
     When invoked as readlink, only the target of the symbolic link is
     printed.  If the given argument is not a symbolic link and the -f option
     is not specified, readlink will print nothing and exit with an error.
```
I missed the "not" when looking up the `-f` option earlier. When rereading this (and the rest of it) the right way, it makes sense - sorry for the confusion.